From eeb9914cd26198da9e2685189f833c5d6a880ff6 Mon Sep 17 00:00:00 2001
From: wangrong <wangrong@uniontech.com>
Date: Mon, 26 Jun 2023 08:08:20 +0000
Subject: [PATCH] fuse: Add setlocale

The gvfsd-ftp will print error messages from gvfs in the debug log.
Without setlocale, the log may contain "???" strings.
---
 client/gvfsfusedaemon.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/client/gvfsfusedaemon.c b/client/gvfsfusedaemon.c
index 1ccef249..79081cdf 100644
--- a/client/gvfsfusedaemon.c
+++ b/client/gvfsfusedaemon.c
@@ -2559,6 +2559,8 @@ main (gint argc, gchar *argv [])
   struct fuse_cmdline_opts opts;
   struct fuse_args args = FUSE_ARGS_INIT (argc, argv);
 
+  setlocale (LC_ALL, "");
+
   if (fuse_opt_parse (&args, NULL, NULL, NULL) == -1)
     return 1;
 
-- 
GitLab
